File: src/main/java/com/smartwallet/market/repository/AssetCategoryRepository.java
package com.smartwallet.market.repository;

import com.smartwallet.market.entity.AssetCategory;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface AssetCategoryRepository extends JpaRepository<AssetCategory, Long> {

    Optional<AssetCategory> findByCode(String code);
    
    List<AssetCategory> findAllByOrderByDisplayOrder();
    
    List<AssetCategory> findByIsActiveTrueOrderByDisplayOrder();
}

File: src/main/kotlin/com/smartwallet/market/repository/AssetCategoryRepository.kt
package com.smartwallet.market.repository

import com.smartwallet.market.model.AssetCategory
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AssetCategoryRepository : JpaRepository<AssetCategory, Long> {
    fun findAllByIsActiveTrueOrderByDisplayOrderAsc(): List<AssetCategory>
    fun findByCode(code: String): AssetCategory?
}
